src/storage-types/src/sinks.rs

```diff
@@ -382,6 +383,36 @@ impl KafkaSinkCompressionType {
     }
 }
 
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Default)]
+pub struct KafkaSinkTopicOptions {
+    /// The replication factor for the topic.
+    /// If `None`, the broker default will be used.
+    pub replication_factor: Option<i32>,
+    /// The number of partitions to create.
+    /// If `None`, the broker default will be used.
+    pub partition_count: Option<i32>,
```

Comment:
What happens if these are `Some(-1)`? It seems like `Option<NonNeg<i32>>` would be safer, in that you don't need to worry about the -1 case representing the same thing as the `None` case.